### DISCUSSION TITLE
Add MariaDB to DatabaseInspection command

### DISCUSSION THREAD

**Discussion by sinnbeck** [laravel/framework#44156] [Ideas]
Currently MariaDB isn't detected when running `php artisan db:show`. This is due to it missing in `DatabaseInspectionCommand::getPlatformName()`. It would be nice to have this added to laravel. It should just be a case of adding these two lines
```
            'MariaDb1027Platform' => 'MariaDB >= 10.2.7',
            'MariaDBPlatform' => 'MariaDB',
```
I will be happy to make a PR if that is wanted.

**Comment by Sophist-UK** [comment]
Here is the [list of platforms in the latest DBAL](https://www.doctrine-project.org/projects/doctrine-dbal/en/latest/reference/platforms.html):
* MySQLPlatform for version 5.0 and above.
* MySQL57Platform for version 5.7 (5.7.9 GA) and above.
* MySQL80Platform for version 8.0 (8.0 GA) and above.
* MariaDb1027Platform for version 10.2 (10.2.7 GA) and above.
* OraclePlatform for all versions.
* SQLServerPlatform for version 2012 and above.
* PostgreSQLPlatform for version 9.4 and above.
* PostgreSQL100Platform for version 10.0 and above.
* Db2Platform for all versions.
* SqlitePlatform for all versions.

The list of [databases supported by Laravel 9](https://laravel.com/docs/9.x/database#introduction) is:
* MariaDB 10.3+
* MySQL 5.7+
* PostgreSQL 10.0+
* SQLite 3.8.8+
* SQL Server 2017+

The list of platforms in `DatabaseInspectionCommand` is currently:
```
'MySQLPlatform' => 'MySQL <= 5',
'MySQL57Platform' => 'MySQL 5.7',
'MySQL80Platform' => 'MySQL 8',
'PostgreSQL100Platform', 'PostgreSQLPlatform' => 'Postgres',
'SqlitePlatform' => 'SQLite',
'SQLServerPlatform' => 'SQL Server',
'SQLServer2012Platform' => 'SQL Server 2012',
```

Comparing these three lists we see that the following changes are needed to `DatabaseInspectionCommand`:
1. Change the `MySQLPlatform` to return `MySQL < 5.7 (unsupported)`;
2. Add `MariaDb1027Platform`  as suggested above;
3. Add `MariaDbPlatform`  to return `MariaDb < 10.2.7 (unsupported)`;
4. Change `SQLServerPlatform` to return `SQL Server > 2012`;
5. Change `SQLServer2012Platform` to return `SQL Server > 2012 (deprecated)`;
6. ~~Fix the bug whereby PostgreSQL100Platform returns itself rather than a string, and add versions to these entries;~~ (my misunderstanding of the match control structure)
7. Add unsupported platforms for Oracle and DB2.

This would make the code as follows:
```
            'MariaDb1027Platform' => 'MariaDB >= 10.2.7',
            'MariaDBPlatform' => 'MariaDB < 10.2.7 (unsupported)',
            'MySQLPlatform' => 'MySQL < 5.7 (unsupported)',
            'MySQL57Platform' => 'MySQL >=5.7 & <8',
            'MySQL80Platform' => 'MySQL >=8',
            'PostgreSQLPlatform' => 'Postgres < 10.0 (unsupported)',
            'PostgreSQL100Platform' => 'Postgres >= 10.0', 
            'SqlitePlatform' => 'SQLite',
            'SQLServerPlatform' => 'SQL Server >= 2012',
            'SQLServer2012Platform' => 'SQL Server >=2012 (deprecated)',
            'Db2Platform' => 'DB2 (unsupported)',
            'OraclePlatform' => 'Oracle (unsupported)',
```

**Comment by Sophist-UK** [comment]
Also, https://github.com/laravel/framework/blob/9.x/src/Illuminate/Database/Console/ShowCommand.php#L141 should be changed from:
```
        $this->components->twoColumnDetail('<fg=green;options=bold>'.$platform['name'].'</>');
```
to
```
        $this->components->twoColumnDetail('<fg=green;options=bold>Platform</>', $platform['name']);
```
in order to format the response correctly.

This changes the output from:
```
  MariaDB >= 10.2.7 ............................................................................................................
  Database ............................................................................................................ database
  Host ............................................................................................................... 127.0.0.1
```
to
```
  Platform ................................................................................................... MariaDB >= 10.2.7
  Database ............................................................................................................ database
  Host ............................................................................................................... 127.0.0.1
```
